lib/missing_step_error.ex

```diff
@@ -0,0 +1,59 @@
+defmodule MissingStepError do
+  @moduledoc """
+  Raises an error, because a feature step is missing its implementation.
+
+  The message of the error will give the user a useful code snippet where
+  variables in feature steps are converted to regex capture groups.
+  """
+
+  defexception [:message]
+
+  @number_regex ~r/(^|\s)\d+(\s|$)/
+  @single_quote_regex ~r/'[^']+'/
+  @double_quote_regex ~r/"[^"]+"/
+
+  def exception(step_text: step_text, step_type: step_type) do
+    converted_step_text =
+      step_text
+      |> convert_nums()
+      |> convert_double_quote_strings()
+      |> convert_single_quote_strings()
+
+    message = """
+    Please add a matching step for:
+    "#{step_type} #{step_text}"
+
+      def#{step_type |> String.downcase} ~r/^#{converted_step_text}$/, vars, state do
+        # Your implementation here
+      end
+    """
+
+    %__MODULE__{message: message}
+  end
+
+  defp convert_nums(step_text) do
+    join_regex_split(Regex.split(@number_regex, step_text), 1, &get_number_string(&1), "")
+  end
+
+  defp convert_double_quote_strings(step_text) do
+    join_regex_split(Regex.split(@double_quote_regex, step_text), 1, &get_double_quote_string(&1), "")
```

Comment:
Can you not use the pipe operator here?

```elixir
@double_quote_regex
|> Regex.split(step_text)
|> join_regex_split(1, &get_double_quote_string(&1), "")
```

It would make this easier to read...